tests/testapp/tests/test_syncsession_methods.py

```diff
@@ -0,0 +1,251 @@
+        rmc = RecordMaxCounter.objects.get(instance_id=self.data['model3_rmc_ids'][0], store_model_id=self.data['model3'])
+        rmcb = RecordMaxCounterBuffer.objects.get(instance_id=self.data['model3_rmcb_ids'][0], model_uuid=self.data['model3'])
+        self.assertNotEqual(rmc.counter, rmcb.counter)
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_ff_rmc(cursor)
+        rmc = RecordMaxCounter.objects.get(instance_id=self.data['model3_rmc_ids'][0], store_model_id=self.data['model3'])
+        rmcb = RecordMaxCounterBuffer.objects.get(instance_id=self.data['model3_rmcb_ids'][0], model_uuid=self.data['model3'])
+        self.assertEqual(rmc.counter, rmcb.counter)
+
+    def test_dequeuing_ff_store(self):
+        store_model = Store.objects.get(id=self.data['model3'])
+        buffer_model = Buffer.objects.get(model_uuid=self.data['model3'])
+        self.assertNotEqual(store_model.last_saved_instance, buffer_model.last_saved_instance)
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_ff_store(cursor)
+        store_model = Store.objects.get(id=self.data['model3'])
+        buffer_model = Buffer.objects.get(model_uuid=self.data['model3'])
+        self.assertEqual(store_model.last_saved_instance, buffer_model.last_saved_instance)
+
+    def test_dequeuing_delete_ff_buffer(self):
+        self.assertTrue(Buffer.objects.filter(model_uuid=self.data['model3']).exists())
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_delete_ff_buffer(cursor)
+        self.assertFalse(Buffer.objects.filter(model_uuid=self.data['model3']).exists())
+
+    def test_dequeuing_delete_ff_rmcb(self):
+        self.assertTrue(RecordMaxCounterBuffer.objects.filter(instance_id=self.data['model3_rmcb_ids'][0], model_uuid=self.data['model3']).exists())
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_delete_ff_rmcb(cursor)
+        self.assertFalse(RecordMaxCounterBuffer.objects.filter(instance_id=self.data['model3_rmcb_ids'][0], model_uuid=self.data['model3']).exists())
+
+    def test_dequeuing_insert_remaining_buffer(self):
+        self.assertFalse(Store.objects.filter(id=self.data['model4']).exists())
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_insert_remaining_buffer(cursor)
+        self.assertTrue(Store.objects.filter(id=self.data['model4']).exists())
+
+    def test_dequeuing_insert_remaining_rmcb(self):
+        for i in self.data['model4_rmcb_ids']:
+            self.assertFalse(RecordMaxCounter.objects.filter(instance_id=i, store_model_id=self.data['model4']).exists())
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_insert_remaining_rmcb(cursor)
+        for i in self.data['model4_rmcb_ids']:
+            self.assertTrue(RecordMaxCounter.objects.filter(instance_id=i, store_model_id=self.data['model4']).exists())
```

Comment:
Also in regards to the comment above, I don't like using `count()` because it could be the case that other records not meant to be in the result set are actually there. So we could have count be 3, but the records are not the 3 we want. So hence the loop is to at least be sure that the records we want to be in there are actually in there. There should be other assertions that assert certain data isn't in there. @jamalex